lib/CMakeLists.txt

```diff
@@ -55,13 +55,18 @@ add_llvm_loadable_module(Svf ${SOURCES})
 add_llvm_Library(LLVMSvf ${SOURCES})
 
 link_directories( ${CMAKE_BINARY_DIR}/lib/Cudd )
-llvm_map_components_to_libnames(llvm_libs bitwriter core ipo irreader instcombine instrumentation target linker analysis scalaropts support )
-target_link_libraries(LLVMSvf ${llvm_libs})
+
+## On Linux platform, we don't link LLVM libs for static/dynamic libraries since we rely on target linkage
+## For Mac, if we don't link we may face errors like "ld: symbol(s) not found for architecture x86_64"
+## need further inspection on this
 if ( CMAKE_SYSTEM_NAME MATCHES "Darwin")
+    llvm_map_components_to_libnames(llvm_libs bitwriter core ipo irreader instcombine instrumentation target linker analysis scalaropts support )
+    target_link_libraries(LLVMSvf ${llvm_libs})
     target_link_libraries(Svf LLVMCudd ${llvm_libs})
 else()
-    target_link_libraries(Svf ${llvm_libs})
+    target_link_libraries(Svf)
 endif()
+set_property(TARGET LLVMSvf PROPERTY POSITION_INDEPENDENT_CODE ON)
```

Comment:
pic will hurt performance. When compiling to SVF executables (e.g., wpa), can we add an option to use the non-pic one for our built-in exes, while exposing to external users for building their own project pic is enabled?